services/web/client/source/class/qxapp/components/widgets/FileManager.js

```diff
@@ -69,14 +51,39 @@ qx.Class.define("qxapp.components.workbench.widgets.FileManager", {
   },
 
   events: {
-    "FileSelected": "qx.event.type.Data",
-    "FolderSelected": "qx.event.type.Data"
+    "ItemSelected": "qx.event.type.Data"
   },
 
   members: {
     __mainTree: null,
     __selectBtn: null,
 
+    _createChildControlImpl: function(id) {
+      let control;
+      switch (id) {
+        case "addButton":
+          control = new qx.ui.form.Button(this.tr("Add file(s)"));
+          this._add(control);
+          break;
+        case "treeMenu":
+          control = new qx.ui.tree.Tree();
+          this._add(control, {
+            flex: 1
+          });
+          break;
+        case "selectButton":
+          control = new qx.ui.form.Button(this.tr("Select"));
+          this._add(control);
+          break;
+        case "progressBox":
+          control = new qx.ui.container.Composite(new qx.ui.layout.HBox());
+          this._addAt(control, 1);
+          break;
+      }
+
+      return control || this.base(arguments, id);
+    },
```

Comment:
except that you now rely on the childern bein instanciated in the right order!

Reply:
Agree, everything is under control though :)